debian/rules

```diff
@@ -2,32 +2,62 @@
 
 DH_VERBOSE = 1
 
-DPKG_EXPORT_BUILDFLAGS = 1
-include /usr/share/dpkg/default.mk
-
+# We want package name to be named snuffleupagus, not php-snuffleupagus
+# even if it doesn't follow Debian PHP module package naming scheme
+# or else we need to review the whole file hierarchy and to generate the
+# package.xml file
+# These two vars have to be set before we include pkg-pecl.mk or else 
+# compilation will fail
+PECL_NAME_OVERRIDE := snuffleupagus
+DH_PHP_VERSIONS_OVERRIDE := $(shell /usr/sbin/phpquery -V)
+include /usr/share/dh-php/pkg-pecl.mk
+
+# make harderning var visible here even if it is defined in pkg-pecl.mk
 export DEB_BUILD_MAINT_OPTIONS = hardening=+all
 
-%:
-	dh $@ 
+# target directory is debian/snuffleupagus, not debian/php-snuffleupagus
+INSTALL_ROOT = $(CURDIR)/debian/$(PECL_NAME)
+
+DIR_TARGETS = $(addprefix build-,$(DH_PHP_VERSIONS))
 
-override_dh_auto_clean:
-	cd ./src; phpize --clean
+## PECL_SOURCE: folder containing source code ex: src
+## SOURCE_DIR: build directory ex: build-7.2
+## PECL_NAME: name of the package
 
-override_dh_auto_configure:
-	cd ./src; phpize
-	cd ./src; ./configure --enable-snuffleupagus
+PECL_SOURCE := src
+CONFIG := config
+$(foreach ver,$(DH_PHP_VERSIONS),$(mkdir build-$(ver)))
```

Comment:
This sounds completely wrong to me: php extensions aren't supposed to work for every versions, only the installed one:

```
$ apt-file list php-ast
php-ast: /etc/php/7.2/mods-available/ast.ini
php-ast: /usr/lib/php/20170718/ast.so
php-ast: /usr/share/doc/php-ast/README.md.gz
php-ast: /usr/share/doc/php-ast/changelog.Debian.gz
php-ast: /usr/share/doc/php-ast/copyright
$ apt-file show php-imagick
php-imagick: /etc/php/7.2/mods-available/imagick.ini
php-imagick: /usr/include/php/20170718/ext/imagick/php_imagick_shared.h
php-imagick: /usr/lib/php/20170718/imagick.so
php-imagick: /usr/share/doc/php-imagick/changelog.Debian.gz
php-imagick: /usr/share/doc/php-imagick/copyright
```

Reply:
`$(foreach ver,$(DH_PHP_VERSIONS)` will compile against only the php7 dev package installed. So if you only installed php7.0 it will compile against this version only.

The good thing to compile against every version is that you get a single package that is compatible with every php7 packages.

Reply:
Yes, this is exactly what I mean: we do *not* want to have a single package for every php version, because this is confusing for system administrators.

[php-libvirt-php]( https://packages.debian.org/sid/amd64/php-libvirt-php/filelist ) is doing it, [php-ds]( https://packages.debian.org/sid/alpha/php-ds/filelist ) is doing it, [xdebug]( https://packages.debian.org/sid/amd64/php-xdebug/filelist ) is doing it, [libsodium]( https://packages.debian.org/sid/amd64/php-libsodium/filelist ) is doing it, … it seems that every single php *library* package in Debian is only supporting a single version of PHP.